### ISSUE TITLE
flatground() do not check collisions with objects

### ISSUE THREAD

**Issue by Otaka** [colobot/colobot#422]
Really do not know if this is a bug, but flatground() function returns that surface is flat regardless if there are any objects on it.
It returns that the surface under my SpaceShip is flat(even under the spaceship ramp). It says to me, that surface under the water also is "flat".

With such issue using this function to determine proper place for automatic building the whole base is not possible.

If astronaut place titanium cube on ground, green dotted circle shows free flat space near the titanium cube and it takes in account collisions with moveable objects(other buildings, robots, ore, titanium cubes). Maybe we should use this procedure for flatground() function, or create new function that expose this functionality?

Colobot 0.1.4-alpha

**Comment by VictorDenisov** [comment]
For free space you need to use function space() instead of flatground().

**Comment by krzys-h** [comment]
@VictorDenisov is correct, you should use space() instead. But we might consider adding another argument to flatground() to make it work like that.

**Comment by Erihel** [comment]
I don't think that adding arument to flatground() with object collision is good. Name of the function is straightforward. Having function do too much is not a good idea.

**Comment by krzys-h** [comment]
Actually, function flatspace() has been already added to do what is requested in this issue. I forgot we had an open issue for that.
By the way, is this function properly documented? @MrSimbax?

**Comment by MrSimbax** [comment]
Could you explain how it was added? Additional argument or does it just check collisions with objects? And why and when this functionality was added, because this issue has "proposal" label and is for milestone 0.2.0, so it might be a little controversial.

**Comment by krzys-h** [comment]
This was added in 0.1.6 along with all the base building functions, for finding place suitable for building. Basically it's a combination of space() with flatground(). This was practially impossible with the existing functions as there is no easy way to get the next position space() would return if the first one is not flat.

I just checked and it is documented properly in SatCom:
![image](https://cloud.githubusercontent.com/assets/1517255/14076779/2d31fc92-f4e5-11e5-9838-183e169d06f6.png)
